hnn_core/gui/_viz_manager.py

```diff
@@ -894,6 +896,61 @@ def _layout_template_change(self, template_type):
             # hide sim-data dropdown
             self.datasets_dropdown.layout.visibility = "hidden"
 
+    def serialize_simulation(self, simulation_name):
+        """
+        Serializes the simulation data for a given simulation name
+        into either a single CSV file
+        or a ZIP file containing multiple CSVs, depending on the number
+        of trials in the simulation.
+        Parameters:
+        - simulation_name (str): The name of the simulation to serialize.
+                                 This name is used to access the corresponding
+                                 data in the 'simulations' dictionary
+                                 of the instance.
+
+        Returns:
+        - tuple: A tuple where the first element is either
+                the CSV content (str) of a single trial
+                or the binary data of a ZIP file (bytes)
+                containing multiple CSV files, and the
+                second element is the file extension (either ".csv" or ".zip").
+        """
+        csv_trials_output = []
+        # CSV file hearders
+        headers = 'times,agg,L2,L5'
+        fmt = '%f, %f, %f, %f'
+
+        for dpl_trial in self.data["simulations"][simulation_name]['dpls']:
+            # Combine all data columns at once
+            signals_matrix = np.column_stack((
+                dpl_trial.times,
+                dpl_trial.data['agg'],
+                dpl_trial.data['L2'],
+                dpl_trial.data['L5']
+            ))
+
+            # Using StringIO to collect CSV data
+            with io.StringIO() as output:
+                np.savetxt(output, signals_matrix, delimiter=',',
+                           header=headers, fmt=fmt)
+                csv_trials_output.append(output.getvalue())
+
+        if len(csv_trials_output) == 1:
+            # Return a single csv file
+            return csv_trials_output[0], ".csv"
+        else:
+            # Create zip file
+            return self._create_zip(csv_trials_output, simulation_name), ".zip"
+
+    def _create_zip(self, csv_data_list, simulation_name):
+        # Zip all files and keep it in memory
+        with io.BytesIO() as zip_buffer:
+            with zipfile.ZipFile(zip_buffer, 'w', zipfile.ZIP_DEFLATED) as zf:
+                for index, csv_data in enumerate(csv_data_list):
+                    zf.writestr(f'{simulation_name}_{index+1}.csv', csv_data)
+            zip_buffer.seek(0)
+            return zip_buffer.read()
+
```

Comment:
I believe these functions are better placed in gui.py as they are not related or dependent on the visualizations.

Reply:
Done!